src/java.desktop/macosx/native/libawt_lwawt/awt/ApplicationDelegate.m

```diff
@@ -520,14 +531,15 @@ + (void)_setDockIconProgress:(NSNumber *)value {
 AWT_ASSERT_APPKIT_THREAD;
 
     ApplicationDelegate *delegate = [ApplicationDelegate sharedDelegate];
-    if ([value doubleValue] >= 0 && [value doubleValue] <=100) {
-        [delegate.fProgressIndicator setDoubleValue:[value doubleValue]];
-        [delegate.fProgressIndicator setHidden:NO];
-    } else {
-        [delegate.fProgressIndicator setHidden:YES];
+    if ([ApplicationDelegate sharedDelegate] != nil) {
```

Comment:
more logical here to check the local variable you just defined.

Reply:
Updated.